packages/sd-jwt/src/action-handler.ts

```diff
@@ -0,0 +1,227 @@
+import { IAgentPlugin } from '@veramo/core'
+import {
+  SdJWTImplementation,
+  ICreateSdJwtVcArgs,
+  ICreateSdJwtVcResult,
+  ICreateSdJwtVcPresentationArgs,
+  ICreateSdJwtVcPresentationResult,
+  IRequiredContext,
+  ISDJwtPlugin,
+  IVerifySdJwtVcArgs,
+  IVerifySdJwtVcResult,
+  IVerifySdJwtVcPresentationArgs,
+  IVerifySdJwtVcPresentationResult,
+  Claims,
+} from './types.js'
+import { mapIdentifierKeysToDocWithJwkSupport } from '@sphereon/ssi-sdk-ext.did-utils'
+import { encodeJoseBlob } from '@veramo/utils'
+
+/**
+ * @beta
+ * SD-JWT plugin for Veramo
+ */
+export class SDJwtPlugin implements IAgentPlugin {
+  readonly schema = schema.ISDJwtPlugin
+
+  constructor(private algorithms: SdJWTImplementation) {}
+
+  // map the methods your plugin is declaring to their implementation
+  readonly methods: ISDJwtPlugin = {
+    createSdJwtVc: this.createSdJwtVc.bind(this),
+    createSdJwtVcPresentation: this.createSdJwtVcPresentation.bind(this),
+    verifySdJwtVc: this.verifySdJwtVc.bind(this),
+    verifySdJwtVcPresentation: this.verifySdJwtVcPresentation.bind(this),
+  }
+
+  /**
+   * Create a signed SD-JWT credential.
+   * @param args - Arguments necessary for the creation of a SD-JWT credential.
+   * @param context - This reserved param is automatically added and handled by the framework, *do not override*
+   * @returns A signed SD-JWT credential.
+   */
+  async createSdJwtVc(args: ICreateSdJwtVcArgs, context: IRequiredContext): Promise<ICreateSdJwtVcResult> {
+    const issuer = args.credentialPayload.iss
+    if (!issuer) {
+      throw new Error('credential.issuer must not be empty')
+    }
+
+    const { alg, key } = await this.getSignKey(issuer, context)
+
+    //TODO: let the user also insert a method to sign the data
+    const signer: Signer = async (data: string) => context.agent.keyManagerSign({ keyRef: key.kid, data })
+
+    const sdjwt = new SDJwtVcInstance({
+      signer,
+      hasher: this.algorithms.hasher,
+      saltGenerator: this.algorithms.saltGenerator,
+      signAlg: alg,
+      hashAlg: 'SHA-256',
+    })
+
+    const credential = await sdjwt.issue(
+      args.credentialPayload,
+      args.disclosureFrame as DisclosureFrame<typeof args.credentialPayload>,
+    )
+    return { credential }
+  }
+
+  /**
+   * Get the key to sign the SD-JWT
+   * @param issuer - did url like did:exmaple.com#key-1
+   * @param context - agent instance
+   * @returns the key to sign the SD-JWT
+   */
+  private async getSignKey(issuer: string, context: IRequiredContext) {
+    const identifier = await context.agent.didManagerGet({
+      did: issuer.split('#')[0],
+    })
+    const doc = await mapIdentifierKeysToDocWithJwkSupport(identifier, 'assertionMethod', context)
+    if (!doc || doc.length === 0) throw new Error('No key found for signing')
+    const key = doc.find((key) => key.meta.verificationMethod.id === issuer)
+    if (!key) throw new Error('No key found with the given id')
```

Comment:
Adjust to our use of accolades and include the issuer value in the error

Reply:
fixed and added